src/Maker/MakeController.php

```diff
@@ -29,6 +31,19 @@
  */
 final class MakeController extends AbstractMaker
 {
+    private $phpCompatUtil;
+
+    public function __construct(FileManager $fileManager, PhpCompatUtil $phpCompatUtil = null)
+    {
+        if (null === $phpCompatUtil) {
+            $phpCompatUtil = new PhpCompatUtil($fileManager);
+
+            trigger_deprecation('symfony/maker-bundle', '1.31', 'Initializing MakeController without providing an instance of PhpCompatUtil is deprecated.');
```

Comment:
This is heavily inspired on https://github.com/symfony/maker-bundle/blob/main/src/Generator.php#L40, tell me if it's not needed in this case

Reply:
I'm not sure if it is worth it. When `maker-bundle` will require PHP 8 as a minimum version, we'll most likely drop the dependency to `PhpCompatUtil`, so in my opinion it's not useful to make this dependency mandatory.

Reply:
No need for the deprecation here. `PhpCompatUtil` is internal & `MakeController` is final - I don't see a use case where someone would use a version >=1.31 and not be able to initiate `MakeController` w/o the utility.

Reply:
Deprecation removed :ok_hand: